Integrating previous appt code into the current codebase

tanweili opened the pull request:
Created and implemented various methods across the classes in the
individual fashion. However, there's a bug where I am unable to view
the appointment I just added. I have to rerun Duke before I am able
to view. I'd appreciate if someone can help me look at the code.

tanweili committed:
Integrating previous appt code into the current codebase

Created and implemented various methods across the classes in the
individual fashion. However, there's a bug where I am unable to view
the appointment I just added. I have to rerun Duke before I am able
to view.

wraineflores commented on src/main/java/seedu/duke/helper/command/ViewAppointmentCommand.java:
```diff
@@ -0,0 +1,17 @@
+        super(parameterArray);
+    }
+
+    public Status execute(List appointmentList) throws NotFoundException {
```
Why is it List instead of AppointmentList?

tanweili replied:
To adhere to the abstract list class methods

wraineflores commented on src/main/java/seedu/duke/helper/command/AddAppointmentCommand.java:
```diff
@@ -0,0 +1,16 @@
+        super(parameterArray);
+    }
+
+    public Status execute(List appointmentList) throws DuplicateEntryException {
```
Why is it List instead of AppointmentList?

wraineflores reviewed (commented):
Tried to trace (Parser -> Command -> Storage -> AddAppointmentCommand -> AppointmentList -> ViewAppointmentCommand -> AppointmentList) why you can't view appointments without re-running Duke but I still don't know why you can't view it after adding

shxr3f commented on src/main/java/seedu/duke/helper/command/ViewAppointmentCommand.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    public Status execute(List appointmentList) throws NotFoundException {
+        String parameters = String.join(",",parameterArray);
```
Will this work if parameterArray is null?

tanweili replied:
It shouldn't be null because in Parser there is 
`    public static Command parseViewAppointment(String parameters) throws HalpmiException {
        String[] viewAppointmentParameters = minParameterCheck(parameters, 2);
        return new ViewAppointmentCommand(viewAppointmentParameters);
    }`
which verifys that it should be length 2

shxr3f commented on src/main/java/seedu/duke/helper/Parser.java:
```diff
@@ -109,14 +100,14 @@ public static Command parseEditMedicine(String parameters) throws HalpmiExceptio
-    public static String[] parseFindAppointment(String parameters) throws HalpmiException {
-        String[] findAppointmentParameters = minParameterCheck(parameters, 2);
-        return findAppointmentParameters;
+    public static Command parseViewAppointment(String parameters) throws HalpmiException {
```
what happens when parameter is null?? eg view appointments

tanweili replied:
It's will throw the HalpMiException stating there is parameters missing

shxr3f reviewed (commented):
I believe you dont consider the case where you dont have any parameters for view appointment command.

tanweili committed:
Resolve bug in previous committed

It was the ordering of nric vs name in constructor method of addAppt.
Also deleted the commented old code in Command.

tanweili committed:
Merge branch 'master' into WeiLi-IntegrateApptCode

shxr3f reviewed (approved):
I think its all good!
